src/generator/AutoRest.Go/CodeNamerGo.cs

```diff
@@ -0,0 +1,491 @@
+        /// <returns>The formatted string.</returns>
+        public override string GetParameterName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                return name;
+            }
+            return EnsureNameCase(GetEscapedReservedName(CamelCase(RemoveInvalidCharacters(name)), "Parameter"));
+        }
+
+        /// <summary>
+        /// Formats a string for naming properties using Pascal case by default.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns>The formatted string.</returns>
+        public override string GetPropertyName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                return name;
+            }
+            return EnsureNameCase(GetEscapedReservedName(PascalCase(RemoveInvalidCharacters(name)), "Property"));
+        }
+
+        /// <summary>
+        /// Formats a string for naming a Type or Object using Pascal case by default.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns>The formatted string.</returns>
+        public override string GetTypeName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                return name;
+            }
+            return EnsureNameCase(GetEscapedReservedName(PascalCase(RemoveInvalidCharacters(name)), "Type"));
+        }
+
+        /// <summary>
+        /// Formats a string for naming a local variable using Camel case by default.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns>The formatted string.</returns>
+        public override string GetVariableName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                return name;
+            }
+            return EnsureNameCase(GetEscapedReservedName(CamelCase(RemoveInvalidCharacters(name)), "Var"));
+        }
+
+        /// <summary>
+        /// Converts names the conflict with Go reserved terms by appending the passed appendValue.
+        /// </summary>
+        /// <param name="name">Name.</param>
+        /// <param name="appendValue">String to append.</param>
+        /// <returns>The transformed reserved name</returns>
+        protected override string GetEscapedReservedName(string name, string appendValue)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException("name");
+            }
+
+            if (appendValue == null)
+            {
+                throw new ArgumentNullException("appendValue");
+            }
+
+            // Use case-sensitive comparisons to reduce generated names
+            if (ReservedWords.Contains(name, StringComparer.Ordinal))
+            {
+                name += appendValue;
+            }
+
+            return name;
+        }
+
+        // Refactor -> Namer
+        public void ReserveNamespace(string ns)
```

Comment:
Another note for future PRs... I don't think the Namespace needs to be added to reserved words. For example, [Container Service SDK](https://github.com/Azure/azure-sdk-for-go/tree/master/arm/containerservice) has a struct named ContainerService (somewhere else in the code we managed to be able to use the namespace for types).
The namespace is only useful in the namer to avoid golint stuttering.